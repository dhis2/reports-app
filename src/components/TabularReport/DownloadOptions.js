import React from 'react'
import PropTypes from 'prop-types'
import i18n from '@dhis2/d2-i18n'
import styles from '../../utils/styles'

const prefix = i18n.t('Download as')

export const DownloadOption = ({ file }) => (
    <a
        style={styles.downloadButton}
        href={file.url}
        download
        tabIndex="0"
        target={file.extension === 'pdf' ? '_blank' : '_self'}
    >
        {prefix} {file.extension}
    </a>
)

DownloadOption.propTypes = {
    file: PropTypes.shape({
        url: PropTypes.string.isRequired,
        extension: PropTypes.string.isRequired,
    }),
}

export const DownloadOptions = ({ fileUrls }) => (
    <div id="download-options-container" style={styles.downloadContainer}>
        {fileUrls.map((file, index) => (
            <DownloadOption file={file} key={index} />
        ))}
    </div>
)

DownloadOptions.propTypes = {
    fileUrls: PropTypes.array.isRequired,
}
