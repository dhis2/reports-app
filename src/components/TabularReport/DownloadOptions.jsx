import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import React from 'react'
import styles from './DownloadOptions.module.css'

const prefix = i18n.t('Download as')

export const DownloadOption = ({ file }) => (
    <a
        className={styles.downloadButton}
        href={file.url}
        download
        tabIndex="0"
        rel="noreferrer"
        target={file.extension === 'pdf' ? '_blank' : '_self'}
    >
        {prefix} {file.extension}
    </a>
)

DownloadOption.propTypes = {
    file: PropTypes.shape({
        extension: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired,
    }),
}

export const DownloadOptions = ({ fileUrls }) => (
    <div className={styles.downloadOptionsContainer}>
        {fileUrls.map((file) => (
            <DownloadOption file={file} key={file.url} />
        ))}
    </div>
)

DownloadOptions.propTypes = {
    fileUrls: PropTypes.array.isRequired,
}
