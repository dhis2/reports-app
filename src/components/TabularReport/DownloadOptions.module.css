.downloadButton {
    cursor: pointer;
    outline: none;
    color: #757575;
    font-size: 12px;
    text-transform: uppercase;
    margin-left: 8px;
}

.downloadOptionsContainer {
    display: flex;
    justify-content: flex-end;
}
