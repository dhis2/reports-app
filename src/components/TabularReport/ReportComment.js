import React from 'react'
import PropTypes from 'prop-types'
import { InputField, Button } from '@dhis2/d2-ui-core'
import i18n from '@dhis2/d2-i18n'

const isCommentingActionEnabled = comment => comment.trim()
const actionButtonPlaceholder = i18n.t('Share')
const inputFieldPlaceholder = i18n.t(
    'Write a comment, question or interpretation of this report'
)

export const ReportComment = props => (
    <div id="share-component">
        <div>
            <InputField
                placeholder={inputFieldPlaceholder}
                type="text"
                multiline
                fullWidth
                value={props.comment}
                onChange={props.setDataSetReportComment}
            />
            <Button
                raised
                color="primary"
                onClick={props.shareDataSetReportComment}
                disabled={!isCommentingActionEnabled(props.comment)}
            >
                {actionButtonPlaceholder}
            </Button>
        </div>
        <style jsx>{`
            div > div {
                margin: 16px 0;
            }
        `}</style>
    </div>
)

ReportComment.propTypes = {
    comment: PropTypes.string.isRequired,
    setDataSetReportComment: PropTypes.func.isRequired,
    shareDataSetReportComment: PropTypes.func.isRequired,
}
