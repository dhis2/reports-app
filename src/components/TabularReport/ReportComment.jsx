import i18n from '@dhis2/d2-i18n'
import { InputField, Button } from '@dhis2/d2-ui-core'
import PropTypes from 'prop-types'
import styles from './ReportComment.module.css'

const isCommentingActionEnabled = (comment) => comment.trim()
const actionButtonPlaceholder = i18n.t('Share')
const inputFieldPlaceholder = i18n.t(
    'Write a comment, question or interpretation of this report'
)

export const ReportComment = (props) => (
    <div id="share-component">
        <div className={styles.wrapper}>
            <InputField
                placeholder={inputFieldPlaceholder}
                type="text"
                multiline
                fullWidth
                value={props.comment}
                onChange={props.setDataSetReportComment}
            />
            <Button
                raised
                color="primary"
                onClick={props.shareDataSetReportComment}
                disabled={!isCommentingActionEnabled(props.comment)}
            >
                {actionButtonPlaceholder}
            </Button>
        </div>
    </div>
)

ReportComment.propTypes = {
    comment: PropTypes.string.isRequired,
    setDataSetReportComment: PropTypes.func.isRequired,
    shareDataSetReportComment: PropTypes.func.isRequired,
}
