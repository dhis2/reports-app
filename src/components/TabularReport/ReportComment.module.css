.wrapper {
    margin: 16px 0;
}
