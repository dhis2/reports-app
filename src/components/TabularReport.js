import React from 'react'
import PropTypes from 'prop-types'
import { DownloadOptions } from './TabularReport/DownloadOptions'
import ReportTable from './TabularReport/ReportTable'
import ReportLoader from './ReportLoader'
import './TabularReport/styles.css'

const TabularReport = ({ content, isLoading, fileUrls }) => (
    <ReportLoader content={content} isLoading={isLoading}>
        <div className="tabular-report">
            <DownloadOptions fileUrls={fileUrls} />
            <ReportTable content={content} />
        </div>
    </ReportLoader>
)

TabularReport.propTypes = {
    content: PropTypes.object.isRequired,
    isLoading: PropTypes.bool.isRequired,
    fileUrls: PropTypes.array.isRequired,
}

export default TabularReport
