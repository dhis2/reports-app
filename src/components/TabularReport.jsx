import PropTypes from 'prop-types'
import React from 'react'
import ReportLoader from './ReportLoader.jsx'
import { DownloadOptions } from './TabularReport/DownloadOptions.jsx'
import ReportTable from './TabularReport/ReportTable.jsx'

const TabularReport = ({ content, isLoading, fileUrls }) => (
    <ReportLoader content={content} isLoading={isLoading}>
        <div className="tabular-report">
            <DownloadOptions fileUrls={fileUrls} />
            <ReportTable content={content} />
        </div>
    </ReportLoader>
)

TabularReport.propTypes = {
    content: PropTypes.object.isRequired,
    fileUrls: PropTypes.array.isRequired,
    isLoading: PropTypes.bool.isRequired,
}

export default TabularReport
