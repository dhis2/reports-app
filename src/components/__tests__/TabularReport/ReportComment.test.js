import React from 'react'
import { shallow } from 'enzyme'
import { ReportComment } from '../../TabularReport/ReportComment'

describe('<ReportComment/>', () => {
    it('Matches the snapshot', () => {
        const props = {
            comment: 'I want to comment on this',
            shareDataSetReportComment: jest.fn(),
            setDataSetReportComment: jest.fn(),
        }
        const wrapper = shallow(<ReportComment {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
