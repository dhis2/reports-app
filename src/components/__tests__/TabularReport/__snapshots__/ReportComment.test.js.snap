// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportComment/> Matches the snapshot 1`] = `
<div
  id="share-component"
>
  <div>
    <TextField
      fullWidth={true}
      multiline={true}
      onChange={[Function]}
      placeholder="Write a comment, question or interpretation of this report"
      type="text"
      value="I want to comment on this"
    />
    <Button
      color="primary"
      disabled={false}
      onClick={[Function]}
      raised={true}
    >
      Share
    </Button>
  </div>
</div>
`;
